File: CSharp/Tests/ReverseInteger.cs
using Solutions;

namespace Tests;

public class ReverseIntegerTest {
    [Fact]
    public void ReverseInteger_Case1() {
        int x = 123;
        int want = 321;

        int result = ReverseInteger.Reverse(x);
        Assert.Equal(want, result);
    }

    [Fact]
    public void ReverseInteger_Case2() {
        int x = -123;
        int want = -321;

        int result = ReverseInteger.Reverse(x);
        Assert.Equal(want, result);
    }

    [Fact]
    public void ReverseInteger_Case3() {
        int x = 120;
        int want = 21;

        int result = ReverseInteger.Reverse(x);
        Assert.Equal(want, result);
    }

    [Fact]
    public void ReverseInteger_Case4() {
        int x = 1534236469;
        int want = 0;

        int result = ReverseInteger.Reverse(x);
        Assert.Equal(want, result);
    }

    [Fact]
    public void ReverseInteger_Case5() {
        int x = -2147483648;
        int want = 0;

        int result = ReverseInteger.Reverse(x);
        Assert.Equal(want, result);
    }

    [Fact]
    public void ReverseInteger_Case6() {
        int x = 900000;
        int want = 9;

        int result = ReverseInteger.Reverse(x);
        Assert.Equal(want, result);
    }
}

File: Rust/tests/reverse_integer.rs
#[cfg(test)]
mod tests {
    use leetcode::reverse_integer::*;

    #[test]
    fn test_reverse_positive_number() {
        let x = 123;
        let want = 321;

        let result = reverse(x);

        assert_eq!(result, want);
    }

    #[test]
    fn test_reverse_negative_number() {
        let x = -123;
        let want = -321;

        let result = reverse(x);

        assert_eq!(result, want);
    }

    #[test]
    fn test_reverse_zero_ending_number() {
        let x = 120;
        let want = 21;

        let result = reverse(x);

        assert_eq!(result, want);
    }

    #[test]
    fn test_reverse_large_number() {
        let x = 1534236469;
        let want = 0;

        let result = reverse(x);

        assert_eq!(result, want);
    }

    #[test]
    fn test_reverse_negative_overflow() {
        let x = -2147483648;
        let want = 0;

        let result = reverse(x);

        assert_eq!(result, want);
    }

    #[test]
    fn test_reverse_positive_overflow() {
        let x = 2147412349;
        let want = 0;

        let result = reverse(x);

        assert_eq!(result, want);
    }

    #[test]
    fn test_reverse_trailing_zeros() {
        let x = 900000;
        let want = 9;

        let result = reverse(x);

        assert_eq!(result, want);
    }
}
